diff --git a/CMakeLists.txt b/CMakeLists.txt
index 36f6587..83470e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -493,7 +482,10 @@ endif()
 
 if(MSVC)
 add_definitions(-D_CRT_SECURE_NO_WARNINGS)
-add_library(poppler STATIC ${poppler_SRCS})
+add_library(poppler SHARED ${poppler_SRCS})
+# so we can determine if to import/export symbols
+target_compile_definitions(poppler PRIVATE _COMPILING_POPPLER=1)
+set_target_properties(poppler PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS true)
 else()
 add_library(poppler ${poppler_SRCS})
 endif()

diff --git a/poppler/GlobalParams.h b/poppler/GlobalParams.h
index 980dc87..1b90301 100644
--- a/poppler/GlobalParams.h
+++ b/poppler/GlobalParams.h
@@ -65,7 +65,14 @@ class SysFontList;
 //------------------------------------------------------------------------

 // The global parameters object.
-extern std::unique_ptr<GlobalParams> POPPLER_PRIVATE_EXPORT globalParams;
+#ifdef _COMPILING_POPPLER
+// cmake's WINDOWS_EXPORT_ALL_SYMBOLS doesn't extend to data unfortunately
+// so we must to this by hand
+extern __declspec(dllexport) std::unique_ptr<GlobalParams> globalParams;
+#else
+extern __declspec(dllimport) std::unique_ptr<GlobalParams> globalParams;
+#endif
+

 //------------------------------------------------------------------------

diff --git a/poppler/PDFDocEncoding.h b/poppler/PDFDocEncoding.h
index 01f3815..278a910 100644
--- a/poppler/PDFDocEncoding.h
+++ b/poppler/PDFDocEncoding.h
@@ -32,7 +32,11 @@

 class GooString;

-extern const Unicode POPPLER_PRIVATE_EXPORT pdfDocEncoding[256];
+#ifdef _COMPILING_POPPLER
+extern __declspec(dllexport) const Unicode pdfDocEncoding[256];
+#else
+extern __declspec(dllimport) const Unicode pdfDocEncoding[256];
+#endif

 char POPPLER_PRIVATE_EXPORT *pdfDocEncodingToUTF16(const std::string &orig, int *length);