diff --git a/glib/CMakeLists.txt b/glib/CMakeLists.txt
index f0aaf99..766223a 100644
--- a/glib/CMakeLists.txt
+++ b/glib/CMakeLists.txt
@@ -121,7 +121,7 @@ if (HAVE_INTROSPECTION AND BUILD_SHARED_LIBS)
 
   # General gir: Reset object-list for introspection & load tool args
   set(INTROSPECTION_GIRS)
-  set(INTROSPECTION_SCANNER_ARGS "--add-include-path=${CMAKE_CURRENT_SOURCE_DIR}" "--warn-all")
+  set(INTROSPECTION_SCANNER_ARGS "--add-include-path=${CMAKE_CURRENT_SOURCE_DIR}" "--warn-all" "--library-path=${CMAKE_CURRENT_BINARY_DIR}" "--library-path=${CMAKE_CURRENT_BINARY_DIR}/..")
   set(INTROSPECTION_COMPILER_ARGS "--includedir=${CMAKE_CURRENT_SOURCE_DIR}")
 
   # Poppler: Assign package to gir & export keys
