diff --git a/poppler/ImageEmbeddingUtils.cc b/poppler/ImageEmbeddingUtils.cc
index ffaa3a9..e451d98 100644
--- a/poppler/ImageEmbeddingUtils.cc
+++ b/poppler/ImageEmbeddingUtils.cc
@@ -14,6 +14,9 @@
 
 #include <config.h>
 
+// conda-forge patch: have to include system headers before JPEG to avoid INT32 redefinitions
+#include "Error.h"
+
 #include <memory>
 #ifdef ENABLE_LIBJPEG
 #    include <cstdio>
@@ -31,7 +34,6 @@ extern "C" {
 #include "goo/GooCheckedOps.h"
 #include "Object.h"
 #include "Array.h"
-#include "Error.h"
 #include "PDFDoc.h"
 
 namespace ImageEmbeddingUtils {
