diff --git a/utils/pdftocairo.cc b/utils/pdftocairo.cc
index 1a0785d..20397b3 100644
--- a/utils/pdftocairo.cc
+++ b/utils/pdftocairo.cc
@@ -50,6 +50,9 @@
 #include <cmath>
 #include <cstring>
 #include <fcntl.h>
+#ifdef _WIN32
+#include <io.h>
+#endif
 #include "parseargs.h"
 #include "goo/gmem.h"
 #include "goo/GooString.h"
@@ -388,8 +391,10 @@ static void writePageImage(GooString *filename)
         return;
 
     if (filename->cmp("fd://0") == 0) {
-#ifdef _WIN32
+#ifdef __CYGWIN__
         setmode(fileno(stdout), O_BINARY);
+#elif _WIN32
+        _setmode(fileno(stdout), O_BINARY);
 #endif
         file = stdout;
     } else
